app/services/send_welcome_email_service.rb

```diff
@@ -1,12 +1,16 @@
 class SendWelcomeEmailService
-  def initialize(mailer:)
-    @mailer = mailer
+  class << self
+    def call(current_user:)
+      new.call(current_user: current_user)
+    end
   end
 
-  def execute(current_user:)
+  def call(current_user:)
     return if current_user.welcome_email_date_utc
 
-    @mailer.send_welcome_email(first_name: current_user.first_name, email: current_user.email).deliver_now
+    WelcomeEmailMailer.
+      send_welcome_email(first_name: current_user.first_name, email: current_user.email).
+      deliver_later(queue: "mailer")
```

Comment:
This might seem like a nitpick but please put the dot operator at the beginning of your lines. It might be easy to misread these as being calls to local methods.